packages/auto-approve/test/auto-approve.test.ts

```diff
@@ -290,4 +298,37 @@ describe('auto-approve', () => {
       });
     });
   });
+
+  describe('gets secrets and authenticates separately for approval', () => {
+    it('creates a separate octokit instance and authenticates with secret in secret manager', async () => {
+      const secretOctokit = new Octokit({auth: '123'});
+      checkPRAgainstConfigStub.returns(true);
+      validateSchemaStub.returns(undefined);
+      checkCodeOwnersStub.returns('');
+      getSecretStub.returns(secretOctokit);
+
+      const payload = require(resolve(
+        fixturesPath,
+        'events',
+        'pull_request_opened'
+      ));
+
+      const scopes = [
+        getConfigFile('fake-config', 200),
+        submitReview(200),
+        addLabels(200),
+        createCheck(200),
+      ];
+
+      await probot.receive({
+        name: 'pull_request.opened',
+        payload,
+        id: 'abc123',
+      });
+
+      scopes.forEach(scope => scope.done());
+      assert.ok(getSecretStub.calledOnce);
+      assert.strictEqual(secretOctokit instanceof TestingOctokit, false);
```

Comment:
I don't understand this assert. As I said in the other comment, can we add a spy on the Octokit returned from `getSecretStub` and make sure the methods are called on that Octokit?

Reply:
@tmatsuo, I don't believe that we need to spy on the Octokit instance, since the test wouldn't be able to make the calls to nock (for submitting the approval) if the `getSecretStub` didn't return with a valid Octokit instance. I definitely agree that this isn't immediately obvious though, and perhaps it points to a need to refactor all the tests to moving to one dependency or another (i.e., sinon > nock). But that might take more work!

This assertion is basically just saying, make sure this Octokit instance is a different type than the one we use to authenticate the bot.

Reply:
This assert is just comparing the local variable `secretOctokit` and the global variable `TestingOctokit`, they are different type from the start and it doesn't ensure anything in the app's code.

To make sure the bot is using the `secretOctokit` correctly, we need to use spy or mock.
If you want, I'll try to come up with the code change and push it to your branch, let me know.